futurex_openedx_extensions/dashboard/serializers.py

```diff
@@ -340,9 +342,63 @@ def _extract_exam_scores(representation_item: dict[str, Any]) -> None:
                 representation_item[possible_key] = score[1] if score else 'no attempt'
 
         representation = super().to_representation(instance)
-
         _extract_exam_scores(representation)
+        return representation
+
+
+class LearnerDetailsForCourseSerializer(
+    LearnerBasicDetailsSerializer, CourseScoreAndCertificateSerializer
+):  # pylint: disable=too-many-ancestors
+    """Serializer for learner details for a course."""
+
+    class Meta:
+        model = get_user_model()
+        fields = LearnerBasicDetailsSerializer.Meta.fields + CourseScoreAndCertificateSerializer.Meta.fields
+
+    def __init__(self, *args: Any, **kwargs: Any):
+        """Initialize the serializer."""
+        super().__init__(*args, **kwargs)
+        self._course_id = CourseLocator.from_string(self.context.get('course_id'))
+        self.collect_grading_info([self._course_id])
+
+    def _get_course_id(self, obj: Any = None) -> CourseLocator:   # pylint: disable=unused-argument
+        """Get the course ID. Its helper method required for CourseScoreAndCertificateSerializer"""
+        return self._course_id
 
+    def _get_user(self, obj: Any = None) -> get_user_model:   # pylint: disable=no-self-use
+        """Get the User. Its helper method required for CourseScoreAndCertificateSerializer"""
+        return obj
+
+
+class LearnerEnrollmentSerializer(CourseScoreAndCertificateSerializer):
+    """Serializer for learner enrollments"""
+    course_id = serializers.SerializerMethodField()
+
+    class Meta:
+        model = CourseEnrollment
+        fields = CourseScoreAndCertificateSerializer.Meta.fields + ['course_id']
+
+    def __init__(self, *args: Any, **kwargs: Any):
+        """Initialize the serializer."""
+        super().__init__(*args, **kwargs)
+        course_ids = self.context.get('course_ids')
+        self.collect_grading_info(course_ids)
+
+    def _get_course_id(self, obj: Any = None) -> CourseLocator | None:   # pylint: disable=no-self-use
+        """Get the course ID. Its helper method required for CourseScoreAndCertificateSerializer"""
+        return obj.course_id if obj else None
+
+    def _get_user(self, obj: Any = None) -> get_user_model | None:   # pylint: disable=no-self-use
+        """Get the User. Its helper method required for CourseScoreAndCertificateSerializer"""
+        return obj.user if obj else None
+
+    def get_course_id(self, obj: Any) -> str:
+        """Get course id"""
+        return str(self._get_course_id(obj))
+
+    def to_representation(self, instance: Any) -> Any:
+        representation = LearnerBasicDetailsSerializer(self._get_user(instance)).data
+        representation.update(super().to_representation(instance))
```

Comment:
why do we merge two serializers here like this? can't we add `LearnerBasicDetailsSerializer` as an inherited parent?

Reply:
I tried to do this way initially, but models are different for both serializers so I was getting issues, I'll try to look into it one more time.

Reply:
thank you for changing the approach in the last push 🌷